src/compiler/emitter.ts

```diff
@@ -3055,17 +3055,34 @@ var __param = (this && this.__param) || function (paramIndex, decorator) {
             }
 
             function emitVariableStatement(node: VariableStatement) {
+                let variableDeclarations = <VariableDeclaration[]>node.declarationList.declarations;
                 let startIsEmitted = true;
-                if (!(node.flags & NodeFlags.Export)) {
-                    startIsEmitted = tryEmitStartOfVariableDeclarationList(node.declarationList);
+                if (node.flags & NodeFlags.Export) {
+                    if (!isES6ExportedDeclaration(node)) {
+                        variableDeclarations = filter(variableDeclarations, node => !!node.initializer);

```

Comment:
I just realized that this implies a new array is constructed every time we emit a variable statement, which might put pressure on the GC. Sorry I didn't catch this a little earlier. 

Instead of a `filter` here's what we can do instead:

* Don't worry about breaking tests for broken code; if it's not syntactically valid JavaScript, then I don't think we're that concerned over broken emit. So first, remove the following:
```TypeScript
if (node.declarationList.declarations.length !== 0 && variableDeclarations.length === 0) {
    return;
}
```
and commit
* Run the tests and accept the baselines. Then add the changed baselines and commit so we can see the differences.
* Write a new function outside of this one named `emitIfInitializerIsPresent` which takes a `VariableDeclaration`. When you're emitting an exported declaration, but you're not in ES6 mode, then instead of `emitCommaList`, we're going to use `emitList` directly (you can omit the null-check in `emitCommaList` in this case).

Let me know if you have any more questions. Again, sorry I didn't catch this before!